src/glados/core/engine.py

```
@@ -124,40 +124,68 @@ class GladosConfig(BaseModel):
-            config = config[key]
+        for path in paths:
+            path = Path(path)
+
+            # Try different encodings
+            for encoding in ["utf-8", "utf-8-sig"]:
+                try:
+                    data = yaml.safe_load(path.read_text(encoding=encoding))
+                    break
+                except UnicodeDecodeError:
+                    if encoding == "utf-8-sig":
+                        raise ValueError(f"Could not decode YAML file {path} with any supported encoding")
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)